src/decider/agent/DeciderAgent.php

```diff
@@ -543,4 +558,103 @@ protected function areOtherScannerFindingsAndRESOAgreed($licenseMatches)
     }
     return true;
   }
-}
+
+  private function getCopyrightsToDisableFalsePositives($uploadId)
+  {
+    if (empty($uploadId)) {
+      return array();
+    }
+    $agentName = 'copyright';
+    $uploadTreeTableName = $this->uploadDao->getUploadtreeTableName($uploadId);
+    $scanJobProxy = new ScanJobProxy($GLOBALS['container']->get('dao.agent'), $uploadId);
+    $scanJobProxy->createAgentStatus(array($agentName));
+    $selectedScanners = $scanJobProxy->getLatestSuccessfulAgentIds();
+    if (!array_key_exists($agentName, $selectedScanners)) {
+      return array();
+    }
+    $latestXpAgentId = $selectedScanners[$agentName];
+    $extrawhere = ' agent_fk='.$latestXpAgentId;
+    $allCopyrights = $this->copyrightDao->getScannerEntries('copyright',
+      $uploadTreeTableName, $uploadId, null, $extrawhere);
+
+    $copyrightJSON = json_encode($allCopyrights);
+    $tmpFile = tmpfile();
+    $tmpFilePath = stream_get_meta_data($tmpFile)['uri'];
+    fwrite($tmpFile, $copyrightJSON);
+    copy($tmpFilePath, '/tmp/copyright.json');
+    $deactivatedCopyrightData = $this->callCopyrightDeactivationClutterRemovalScript($tmpFilePath, $clutter_flag = 0);
+    if (!$deactivatedCopyrightData) {
+      return array();
+    }
+    $deactivatedCopyrights = json_decode($deactivatedCopyrightData, true);
+    foreach ($deactivatedCopyrights as $deactivatedCopyright) {
+      $item = $deactivatedCopyright['uploadtree_pk'];
+      $itemTreeBounds = $this->uploadDao->getItemTreeBounds($item, $uploadTreeTableName);
+      $hash = $deactivatedCopyright['hash'];
+      $content = $deactivatedCopyright['content'];
+      $cpTable = 'copyright';
+      if ($deactivatedCopyright['is_copyright'] == "t") {
+        $action = '';
+        $hash = '';
+      } else {
+        $action = 'delete';
+      }
+      $this->copyrightDao->updateTable($itemTreeBounds, $hash, $content, $this->userId, $cpTable, $action);
+      $this->heartbeat(1);
+    }
+    fclose($tmpFile);
+  }
+  private function getCopyrightsToDisableFalsePositivesClutter($uploadId)
```

Comment:
Both functions `getCopyrightsToDisableFalsePositives()` and `getCopyrightsToDisableFalsePositivesClutter()` differ only while calling `callCopyrightDeactivationClutterRemovalScript()` with flag `0` or `1` and while setting the values for **hash** and **content**. Can these be merged?

```diff
diff --git a/onlyfp b/fpandc
index baace11..9601d0b 100644
--- a/onlyfp
+++ b/fpandc
@@ -27,13 +27,13 @@
     foreach ($deactivatedCopyrights as $deactivatedCopyright) {
       $item = $deactivatedCopyright['uploadtree_pk'];
       $itemTreeBounds = $this->uploadDao->getItemTreeBounds($item, $uploadTreeTableName);
-      $hash = $deactivatedCopyright['hash'];
-      $content = $deactivatedCopyright['content'];
       $cpTable = 'copyright';
       if ($deactivatedCopyright['is_copyright'] == "t") {
         $action = '';
         $hash = '';
+        $content = $deactivatedCopyright['edited_text'];
       } else {
+        $hash = $deactivatedCopyright['hash'];
         $action = 'delete';
       }
       $this->copyrightDao->updateTable($itemTreeBounds, $hash, $content, $this->userId, $cpTable, $action);
```